rtCamp/privacy-sandbox-demos
# Implement Storage Access API Demo

## Description 
This pull request adds the new Storage Access API Demo to demonstrate one of the solutions to use third-party cookies after the 3PCD. This demo replicates the[ e-commerce cross-domain shopping cart](https://github.com/GoogleChromeLabs/ps-analysis-tool/wiki/Example-analysis-scenarios#e-commerce-cross-domain-shopping-cart) scenario. 

## Relevant Technical Choices
- implement the verification for storage access API
- check if the content has been loaded from an iframe or top-level
- reuse the same logic from the e-commerce demo to fetch the data from the cart and add a new item

## Testing instructions
1. Clone the repository and checkout `feature/adding-storage-access-api` branch containing these changes.
2. Ensure you have Node.js and npm installed.
3. Run `npm install` to install all dependencies.
4. Set up the `.env` file.
5. Start the server using `npm start`.
6. Navigate to the homepage and click on the "Storage Access API" link.
7. Open the top-level site and add a product
8. Go back to the Storage Access API demo, reload the page
9. Check if the button "Load Cart" is visible
10. Grant access to the Storage Access API
11. Click on the "Load Cart Button"
12. Add products to the cart and check if the counter is increasing

#Screenshots

**Storage Access API page before grant access**
![Screenshot 2024-01-16 at 18 40 47](https://github.com/rtCamp/privacy-sandbox-demos/assets/330792/5cdc56a7-768e-471e-a23d-c98b89ffd300)

**Permission pop-up**
![Screenshot 2024-01-16 at 18 41 03](https://github.com/rtCamp/privacy-sandbox-demos/assets/330792/856ee15d-7121-4066-a653-58e85915438c)


**Storage Access API page after grant access**
![Screenshot 2024-01-16 at 18 41 12](https://github.com/rtCamp/privacy-sandbox-demos/assets/330792/01f728ef-6b89-440c-8f9b-080c19e9ef03)
